cmake/functions.cmake

```diff
@@ -30,9 +30,7 @@ endfunction()
 function(AddTest test_name SOURCES)
   list(APPEND SOURCES main.cc)
   add_executable(${test_name} ${SOURCES})
-  add_dependencies(${test_name} gtest)
-  target_include_directories(${test_name} PRIVATE ${gtest_INCLUDE_DIR})
-  target_link_libraries(${test_name} PRIVATE ${gtest_LIBRARIES} pthread)
+  target_link_libraries(${test_name} PRIVATE gtest)
```

Comment:
Both [benchmarks](https://github.com/google/benchmark/blob/master/src/CMakeLists.txt#L23) and [tests](https://github.com/google/googletest/blob/48ee8e98abc950abd8541e15550b18f8f6cfb3a9/googletest/cmake/internal_utils.cmake#L161) are already linked to pthread.